ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -43,10 +46,20 @@ var (
 	PrioritizeWorkersToDelete bool
 	ForcedClusterUpgrade      bool
 	EnableBatchScheduler      bool
+
+	// Definition of a index field for pod name
+	podUIDIndexField = "metadata.uid"
 )
 
 // NewReconciler returns a new reconcile.Reconciler
 func NewReconciler(mgr manager.Manager) *RayClusterReconciler {
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Pod{}, podUIDIndexField, func(rawObj client.Object) []string {
+		pod := rawObj.(*corev1.Pod)
```

Comment:
The following change may fix https://github.com/ray-project/kuberay/issues/601#issuecomment-1486149187. cc @jeevb 

```
pod := rawObj.(*corev1.Pod)
owner := metav1.GetControllerOf(pod)
if owner == nil || owner.APIVersion != rayiov1alpha1.GroupVersion.String() || owner.Kind != "RayCluster" {
	return nil
}
return []string{string(pod.UID)}
```

Reply:
^ 

Above is wrong. This is used to enable efficient lookup operations for the local cache. The informer will still fetch cluster status from Kubernetes APIServer. That is, events which are unrelated to Ray Pods will finally be fetched by the informer. Sorry for the misleading.